sky/engine/core/text/Paragraph.cpp

```diff
@@ -107,4 +110,33 @@ void Paragraph::paint(Canvas* canvas, const Offset& offset)
     skCanvas->translate(-offset.sk_size.width(), -offset.sk_size.height());
 }
 
+std::vector<Rect> Paragraph::getRectsForRange(unsigned start, unsigned end) {
+  if (end <= start || start == end)
+    return std::vector<Rect>();
+
+  unsigned offset = 0;
+  Vector<IntRect> rects;
+  for (RenderObject* object = m_renderView.get(); object; object = object->nextInPreOrder()) {
+    if (!object->isText())
```

Comment:
we probably want to show inline replaced boxes (e.g. images) as selected too (once we have a way to have "replaced boxes")